### PR TITLE
web: Support ExternalInterface.call("function(){...}")

### PR DESCRIPTION
**Pull request by Dinnerbone**
Plus many tests for ExternalInterface on web

~~Depends on #16500 (drafted until that's merged)~~

Progresses #8330 (I assume)
Progresses #15013

### CODE REVIEW

**Comment by sammypanda** [comment]
this is awesome (^^) thank u :heart:

**Comment by evilpie** [comment]
I am not sure if this would pass the add-on review for addons.mozilla.org.

**Comment by Dinnerbone** [comment]
> I am not sure if this would pass the add-on review for addons.mozilla.org.

Hmm, that's unfortunate. Do you know which guideline this would be violating and how we can fix it?

**Comment by evilpie** [comment]
You are definitely going to get an automatic warning message. I think it might be possible to explain why the `eval` exists and why it is necessary. Some other extensions like content script messengers also use something like `eval` and can get accepted. Technically running Flash content is like eval but with extra steps :smile:

**Comment by torokati44** [comment]
Given the risk factors mentioned above, and experienced elsewhere, shall we wait with this until the next AMO submission at least?

**Comment by wsxarcher** [comment]
It also fix
https://github.com/ruffle-rs/ruffle/issues/15009
https://github.com/ruffle-rs/ruffle/issues/16482

File: web/packages/core/src/ruffle-imports.ts [hunk 1]
```diff
@@ -61,12 +61,15 @@ export function copyToAudioBufferInterleaved(
...
  */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function getProperty(target: any, key: string): any {
-    return target[key];
+// @ts-expect-error defined but not used
+// eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/no-unused-vars
+export function callExternalInterface(name: string, args: any[]): any {
+    // [NA] Yes, this is direct eval. Yes, this is a Bad Thing when it comes to security.
+    // In fact, yes this is vulnerable to an XSS attack!
+    // But plot twist: Flash allowed for this and many content *relies on it*. :(
```
Comment (Lord-McSweeney) [inline web/packages/core/src/ruffle-imports.ts:73]: "many content" typo?

Comment (Dinnerbone) [reply web/packages/core/src/ruffle-imports.ts:73]: I don't think so?

Comment (danielhjacobs) [reply web/packages/core/src/ruffle-imports.ts:73]: I'd say much content or many types of content (or some other grouping), content is treated as an uncountable noun. Alternatively, a lot of content.

**Comment by torokati44** [comment]
Last week's AMO submission has been accepted, so apparently this didn't trip any extra alarms during review ... so far.

**Pull request merged**